catalog/plugin/catalog-plugin-metacard-validation/src/main/java/ddf/catalog/metacard/validation/MetacardValidityFilterPlugin.java

```diff
@@ -0,0 +1,95 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostQueryPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+public class MetacardValidityFilterPlugin implements PostQueryPlugin {
+
+    private static final Logger LOGGER = LoggerFactory
+            .getLogger(MetacardValidityFilterPlugin.class);
+
+    private static Map<String, List<String>> attributeMap = new HashMap<>();
+
+    public static final String INVALID_STATE = "invalid_state";
+
+    @Override
+    public QueryResponse process(QueryResponse input)
+            throws PluginExecutionException, StopProcessingException {
+        List<Result> results = input.getResults();
+
+        // process security
+        for (Result result : results) {
+            Metacard metacard = result.getMetacard();
+            Attribute securityAttribute = metacard.getAttribute(Metacard.SECURITY);
+            HashMap<String, List<String>> securityMap = null;
+            Serializable tempSecurityMap = null;
+            if (securityAttribute != null && securityAttribute.getValue() != null && securityAttribute.getValue() instanceof Map) {
+                securityMap = (HashMap<String, List<String>>) securityAttribute.getValue();
+            } else {
+                securityMap = new HashMap<>();
+            }
+            if ((metacard.getAttribute(VALIDATION_ERRORS) != null && metacard.getAttribute(VALIDATION_ERRORS).getValues() != null)
+                    || (metacard.getAttribute(VALIDATION_WARNINGS) != null && metacard.getAttribute(VALIDATION_WARNINGS).getValues() != null)) {
+                for (Map.Entry<String, List<String>> attributeMapping : attributeMap
+                        .entrySet()) {
+                    securityMap.put(attributeMapping.getKey(), attributeMapping.getValue());
+                }
+
+            }
+            if (!securityMap.isEmpty()) {
+                metacard.setAttribute(new AttributeImpl(Metacard.SECURITY, securityMap));
+            }
+
+        }
+        return input;
+    }
+
+    public Map<String, List<String>> getAttributeMap() {
+        return attributeMap;
+    }
+
+    public void setAttributeMap(List<String> attributeMappings) {
+        for (String attributeMapping : attributeMappings) {
+            String[] keyValue = attributeMapping.split("=");
+            attributeMap.put(keyValue[0].trim(),
+                    Arrays.asList(keyValue[1].split(",")).stream().map(String::trim)
+                            .collect(Collectors.toList()));
+        }
+    }
+
+    public void setAttributeMap(Map<String, List<String>> attributeMap) {
```

Comment:
The setter with the`Map<String, List<String>>` is required for blueprint - I had left it out at first but the blueprint failed to come up when I didn't have it.